### DISCUSSION TITLE
Bipartite graph target node classification

### DISCUSSION THREAD

**Discussion by entinsd** [pyg-team/pytorch_geometric#5933] [Q&A]
I have a custom bipartite graph dataset with Users and Websites and edges with weights indicating session count. I want to classify websites into two classes (Suspicious/Not Suspicious), have labels in the dataset. I'm trying to implement this classification model with PyTorch Geometric and facing difficulty with building a custom PyG dataset(HeteroData) having labels to predict only for Websites. And which GNN model to use (SAGEConv)? Is there any example or tutorial with a similar problem implementation?

**Comment by wsad1** [comment]
You don't need a `HeteroData` object for bipartite graphs. GNN's like `SAGEConv` can handle bipartite graphs.
```
users # torch tensor of shape num_users*num_user_feats
website # torch tensor of shape num_websites * num_website_feats
edge_index # tensor representing edges going from user to website
conv = SAGEConv((num_user_feats, num_website_feats), out_dim)
out = conv((user, website), ege_index) # out can be used to classify websites
```
